### ISSUE TITLE
v2 output shifted when using powerline

### ISSUE THREAD

**Issue by zeus86** [chubin/wttr.in#345]
i am using the kde-konsole on ubuntu 18.04 in this case with powerline-shell (https://github.com/b-ryan/powerline-shell), and a patched font for powerline (hack.ttf). In this case, some icons are moved to the right.
![wttr](https://user-images.githubusercontent.com/11415964/64417826-148f7e00-d09a-11e9-8a0c-f6f5c1c820f7.png)

**Comment by chubin** [comment]
To find out what is the width of different icons, we need some more data.
Could you please run the following command:
```
for i in Dortmund Paris Berlin Moscow London Amsterdam Madrid; do curl -ks v2.wttr.in/$i | sed -n 30p; done
```
that will produce different lines with different weather conditions
and so it would be possible to find out the width of each character.

**Comment by zeus86** [comment]
sure, here you have it:
![wttr2](https://user-images.githubusercontent.com/11415964/64422939-89b48080-d0a5-11e9-853e-b5412c4c69cc.png)

**Comment by chubin** [comment]
@zeus86 Thank you! Is there any line that has the correct width?

**Comment by zeus86** [comment]
> @zeus86 Thank you! Is there any line that has the correct width?
...compared with the rest of the normal output: no. the 5th output line  in the script output above (so the output from "London") has almost the right size of the rest of the bordered box, but the delimiter is shifted inwards onto the weather icons. 
but in short: no, no line has an output matching the rest of the box.

**Comment by zeus86** [comment]
a quick note to this: I tested the one above on my work-notebook. it runs on Ubuntu 18.04. On my other notebook with arch installed, it shows just fine, even with powerline installed and active. on my girlfriends computer with ubuntu 18.04 without powerline it is also broken, so this error does not refer to powerline, but to ubuntu!

on my arch-machine the dotted line is awkwardly drawn, the color is correct, (from red==high to light blue==low, but the graphic looks like wrong encoding (blocky chars instead of dots). Also some weather icons are small then, but maybe this is a font issue. On arch I use the font "Hack" in 9pt with solarized dark scheme.
the screenshot below is from my arch machine. 

![wttr-arch](https://user-images.githubusercontent.com/11415964/64480247-4b61b300-d1c4-11e9-9ddc-ca9faf16fc66.png)

**Comment by jmcastagnetto** [comment]
Just a FYI. Works OK using Terminator + Powerline fonts in Ubuntu 18.04 64-bit. Seems to work in Gnome Terminal too.

Perhaps the issue is with Konsole? (I don't use it, so haven't tested it)

![screenshot-fb185b22-91df-4fb2-8577-08d72a1e9114-2019 11 12-12_50_39](https://user-images.githubusercontent.com/364668/68696752-8fb65a00-054b-11ea-9492-f57beee9a602.png)

**Comment by claenn** [comment]
Output in @iSH_app, iPadOS14.1, monospaced Menlo

![IMG_1181](https://user-images.githubusercontent.com/8040019/97639664-4ce0e100-1a3f-11eb-936a-8f7444be9ef4.jpeg)

**Comment by chubin** [comment]
@claenn thank you

**Comment by lxcode** [comment]
I believe that the issue is with WEATHER_SYMBOL_WIDTH_VTE treating some emoji like the plain sun and cloud graphemes as single-width and adding an extra space, while since Unicode 9 all emoji are supposed to be considered double-width.

So if this were to be configurable, it should just be between the current behavior (which I guess was designed to make some version of VTE happy) and simply treating everything as double-width (the modern "correct" way). I'd argue the default should be the latter.

**Comment by chubin** [comment]
@lxcode 

I think we need an option to make it configurable, at least until the point when all possible terminals will have the same behavior (which I don't expect to come in any foreseeable future).

My idea is to sort all emojis, and encode each emoji in the list with 0 (for 1-char width) or 1 (for 2-char width), so it will be a binary vector, then it can be converted into a two-byte value (because we have <= 16 different emojis).

The query will look like this:

```
curl v2.wttr.in?emoji_width=0x1234
```

In your case it would be something like `0xffff` (all emojis are 2-charachter width).

For known terminals, we could make a table with recommended values,
and some special mode to find the code for the current terminal if it is not known.

What do you think?

**Comment by lxcode** [comment]
Hm, that does seem a reasonable way to do full configurability. 

As far as automatically determining the widths, I imagine some combination of jquast's ucs-detect/[wcwidth](https://github.com/jquast/wcwidth) could work, though it's possible that there could be some trip-ups with various fonts.

**Comment by chubin** [comment]
@lxcode I think it depends on the terminal, and `wcwidth` can't know it. Maybe there is some system behind it (even I am absolutely sure, it should be), but I didn't manage to find it so far. Maybe, as soon as this feature is implemented, we could categorize the terminals and find a generalized approach

**Comment by lxcode** [comment]
Yeah you're right, I was thinking of a locally running program for some reason, which this of course isn't. Sorry!

**Comment by chubin** [comment]
I think even if it runs locally, it can't be sure about each terminal. If I understand it right, it has its own idea about each emoji width, and it does not always correspond to all terminals

**Comment by chubin** [comment]
Also, this thread https://x.com/Dewry/status/1982822274173370514 by @wind0204

**Comment by chubin** [comment]
In the new Go implementation (#1225), we have finally taken a step toward solving the problem:

Now it is possible to select the emoji profile (option `emoji`), which also affects emoji width.

This can help in finding the best width for the current terminal.

For example, here we have two different terminals (Alacritty and Ghostty), which render emojis differently, including with varying widths. However, with the new parameter, the view can be rendered correctly in both cases.

**Alacritty**:    
(`emoji=narrow`)

<img width="1118" height="1337" alt="Image" src="https://github.com/user-attachments/assets/658f4db9-27fa-45ca-a66d-2d85bfa6d959" />

&nbsp;

&nbsp;

**Ghostty**:
(`emoji=unicode`, default)

<img width="1273" height="1637" alt="Image" src="https://github.com/user-attachments/assets/a0c247e9-085b-4127-9791-a7a10dbb05d0" />

&nbsp;

**Test please**

If by chance is anybody here still alive + using v2.wttr.in => please test and let me know if it works.

**Comment by bonswouar** [comment]
I can confirm the default unicode version seems also to work pretty well with foot now:

<img height="500" alt="Image" src="https://github.com/user-attachments/assets/daeb2412-facc-40fa-894e-b81a135b9b3a" />
